production/helm/loki/README.md

```diff
@@ -25,6 +25,119 @@ Add the following repo to use the chart:
 helm repo add grafana https://grafana.github.io/helm-charts
 ```
 
+## Upgrading from v2.x
+
+v3.x represents a major milestone for this chart, showing a committment by the Loki team to provide a better supported, scalable helm chart.
+In addition to moving the source code for this helm chart into the Loki repo itself, it also combines what were previously two separate charts,
+[`grafana/loki`](https://github.com/grafana/helm-charts/tree/main/charts/loki) and [`grafana/loki-simple-scalable`](https://github.com/grafana/helm-charts/tree/main/charts/loki-simple-scalable) into one chart. This chart will automatically assume the "Single Binary" mode previously deployed by the `grafana/loki` chart if you are using a filesystem backend, and will assume the "Scalable" mode previoulsy deployed by the `grafana/loki-simple-scalable` chart if you are using an object storage backend.
+
+As a result of this major change, upgrades from the charts this replaces might be difficult. We are attempting to support the 3 most common upgrade paths.
+
+  1. Upgrade from `grafana/loki` using local `filesystem` storage
+  1. Upgrade from `grafana/loki` using a cloud based object storage such as S3 or GCS, or an api compatible equivilent like MinIO.
+  1. Upgrade from `grafana/loki-simple-scalable` using a cloud based object storage such as S3 or GCS, or an api compatible equivilent like MinIO.
+
+### Upgrading from `grafana/loki`
+
+#### Upgrading from `filesystem` storage
+
+When Loki is backed by `filesystem` storage, we assume a single instance of Loki that is not HA. As a result, this upgrade method will involve downtime. The upgrade will involve deleting the previously deployed loki stateful set, the running the `helm upgrade` which will create the new one with the same name, which should attach to the existing PVC or ephemeral storage, thus preserving you data. Will still highly recommend backing up all data before conducting the upgrade.
+
+To upgrade, you will need at least the following in your `values.yaml`:
+
+```yaml
+upgradeFromV2: true,
+loki:
+  commonConfig:
+    replication_factor: 1
+  storage:
+    type: 'filesystem'
+```
+
+Assuming you installed `grafana/loki` as `loki` in namespace `loki`, your upgrade command might look like:
+
+```console
+helm upgrade loki ~/workspace/grafana/loki/production/helm/loki --values environments/loki-migration-test/values-new.yaml --namespace loki
+```
+
+You will need to manually delete the existing stateful set for the above command to work.
```

Comment:
Could you give the `kubectl` command for this?

Reply:
the assumption is you are using `helm`, thus there is no `kubectl` command. If a user want to run `helm template` to `kubectl` the raw yaml themselves I would consider that an advanced technique we don't need to document.

Reply:
ahh, yes, haha, and I suppose hit `<ctrl-d>` in `k9s` isn't appropriate, even though it's what I always use now so I've forgotten the `kubectl` command